test/Feature/SingularSpec.hs

```diff
@@ -79,44 +101,68 @@ spec =
         p <- request methodPost
           "/addresses"
           [("Prefer", "return=representation"), singular]
-          [json| [ { id: 100, address: "xxx" } ] |]
-        liftIO $ simpleBody p `shouldBe` [str|{"id":100,"address":"xxx"}|]
+          [json| [ { id: 102, address: "xxx" } ] |]
+        liftIO $ simpleBody p `shouldBe` [str|{"id":102,"address":"xxx"}|]
 
       it "works for one row even with return=minimal" $ do
         request methodPost "/addresses"
           [("Prefer", "return=minimal"), singular]
-          [json| [ { id: 101, address: "xxx" } ] |]
+          [json| [ { id: 103, address: "xxx" } ] |]
           `shouldRespondWith` ""
           { matchStatus  = 201
           , matchHeaders = ["Content-Range" <:> "*/*"]
           }
         -- and the element should exist
-        get "/addresses?id=eq.101"
-          `shouldRespondWith` [str|[{"id":101,"address":"xxx"}]|]
+        get "/addresses?id=eq.103"
+          `shouldRespondWith` [str|[{"id":103,"address":"xxx"}]|]
           { matchStatus  = 200
           , matchHeaders = []
           }
 
       it "raises an error when attempting to create multiple entities" $ do
         p <- request methodPost
           "/addresses"
-          [("Prefer", "return=representation"), singular]
+          [singular]
           [json| [ { id: 200, address: "xxx" }, { id: 201, address: "yyy" } ] |]
         liftIO $ simpleStatus p `shouldBe` notAcceptable406
 
         -- the rows should not exist, either
         get "/addresses?id=eq.200" `shouldRespondWith` "[]"
 
-      it "return=minimal allows request to create multiple elements" $
+      it "raises an error when attempting to create multiple entities with return=rep" $ do
+        p <- request methodPost
+          "/addresses"
+          [("Prefer", "return=representation"), singular]
+          [json| [ { id: 202, address: "xxx" }, { id: 203, address: "yyy" } ] |]
+        liftIO $ simpleStatus p `shouldBe` notAcceptable406
+
+        -- the rows should not exist, either
+        get "/addresses?id=eq.202" `shouldRespondWith` "[]"
+
+      it "singular behavior holds priority over return=minimal" $ do
```

Comment:
Now that I cleared my doubt, I think it's not about priority but more about consistent behavior. `singular` will enforce behavior regardless of `return=minimal` or `return=rep`.

So maybe this test case can be rephrased as `it "raises an error regardless of return=minimal"`.